TestResultSummaryService/parsers/BenchmarkParser.js

```diff
@@ -46,116 +49,78 @@ class BenchmarkParser extends Parser {
-                            continue;
-                        }
-                    }
-
-                    // multiple metric values are present
-                    if (BenchmarkMetric[curBenchmarkNamedRouted]["metrics"][i].regexRepeat === true) {
-                        curMetricValue = [];
-                        curRegexResult = BenchmarkMetric[curBenchmarkNamedRouted]["metrics"][i].regex.exec( curSearchString )
-
-                        while (curRegexResult !== null ) {
-                            curMetricValue.push(parseFloat(curRegexResult[1]));
-                            curRegexResult = BenchmarkMetric[curBenchmarkNamedRouted]["metrics"][i].regex.exec( curSearchString );
-                        }
-
-                    // single metric value
-                    } else {
-                        if ( ( curRegexResult = BenchmarkMetric[curBenchmarkNamedRouted]["metrics"][i].regex.exec( curSearchString ) ) !== null ) {
-                            curMetricValue = [parseFloat(curRegexResult[1])];
-                        }
-                    }
-
-                    curTestData["metrics"].push({name: curMetric, value: curMetricValue})
-                }
-
-                // Parse Java version
-                if ( ( curRegexResult = javaVersionRegex.exec( curItr.value ) ) !== null ) {
-                    curJavaVersion = curRegexResult[1];
-                } else {
-                    curJavaVersion = "benchmarkProduct: " + curBenchmarkProduct;
-                }
-                curTestData["javaVersion"] = curJavaVersion;
-
-                if ( ( curRegexResult = javaBuildDateRegex.exec( curJavaVersion ) ) !== null ) {
-                    curJavaBuildDate = curRegexResult[1];
-                }
-                curTestData["jdkBuildDateUnixTime"] = this.convertBuildDateToUnixTime( curJavaBuildDate );
-            }
-
-            tests.push( {
-                testOutput: curItr.value,
-                testResult: isValid ? "PASSED" : "FAILED",
-                testIndex: testIndex++,
-                benchmarkName: curBenchmarkName,
-                benchmarkVariant: curBenchmarkVariant,
-                benchmarkProduct: curBenchmarkProduct,
-                testData: curTestData
-            } );
+            let curBenchmarkVariant = benchmarkVariantRegex.exec( curItr.value )[1];
+            
+            let curBenchmarkProduct = benchmarkProductRegex.exec( curItr.value )[1];
+             
+            let curBenchmarkNamedRouted = BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant];
+    
+            curTestData["javaVersion"] = javaVersionRegex.exec( curItr.value )[1];
+            
+            curTestData["jdkBuildDateUnixTime"] = javaBuildDateRegex.exec( curTestData["javaVersion"] )[1];
+            
+              
+   
+            // parse metrics
+            curTestData['metrics'] = [];
+            for ( let i = 0; i < BenchmarkMetric[curBenchmarkNamedRouted]['metrics'].length; i++ ) 
```

Comment:
We are using `BenchmarkMetric[curBenchmarkNamedRouted]['metrics']` 3 times so maybe we can use a variable instead.